File: src/sentry/api/serializers/models/sentry_app_webhook_error.py
from __future__ import absolute_import

from sentry.api.serializers import Serializer, register
from sentry.models import SentryAppWebhookError


@register(SentryAppWebhookError)
class SentryAppWebhookErrorSerializer(Serializer):
    def serialize(self, obj, attrs, user):
        data = {
            "app": {
                "uuid": obj.sentry_app.uuid,
                "slug": obj.sentry_app.slug,
                "name": obj.sentry_app.name,
            },
            "date": obj.date_added,
            "organization": {"slug": obj.organization.slug, "name": obj.organization.name},
            "errorId": obj.error_id,
            "request": {"body": obj.request_body, "headers": obj.request_headers},
            "eventType": obj.event_type,
            "webhookUrl": obj.webhook_url,
            "response": {"body": obj.response_body, "statusCode": obj.response_code},
        }

        return data

File: tests/js/sentry-test/fixtures/sentryAppWebhookError.js
export function SentryAppWebhookError(params = {}) {
  return {
    app: {
      slug: 'sample-app',
      name: 'Sample App',
      uuid: 'b468fed3-afba-4917-80d6-bdac99c1ec05',
    },
    date: '2019-09-25T23:54:54.440Z',
    organization: {
      slug: 'test-org',
      name: 'Test Org',
    },
    errorId: null,
    request: {
      body: {},
      headers: {},
    },
    eventType: 'issue.assigned',
    webhookUrl: 'https://example.com/webhook',
    response: {
      body: 'This is an error',
      statusCode: 400,
    },
    ...params,
  };
}
